server/src/main/scala/com/cloudera/livy/server/WebServer.scala

```diff
@@ -42,11 +42,14 @@ class WebServer(livyConf: LivyConf, var host: String, var port: Int) extends Log
   server.setStopTimeout(1000)
   server.setStopAtShutdown(true)
 
+  var protocol = "http"
```

Comment:
You could also make this a `val` (which is preferable) by changing the code that initializes `connector`:

```
val (connector, protocol) = ... {
  case None => (new ServerConnector(server), "http")
  ...
}
```